src/query/sql/src/planner/optimizer/property/selectivity.rs

```diff
@@ -292,14 +289,8 @@ impl<'a> SelectivityEstimator<'a> {
             if !self.updated_column_indexes.contains(index) {
                 let new_ndv = (column_stat.ndv * selectivity).ceil();
                 column_stat.ndv = new_ndv;
-                if let Some(histogram) = &mut column_stat.histogram {
-                    if new_ndv as u64 <= 2 {
-                        column_stat.histogram = None;
-                    } else {
-                        for bucket in histogram.buckets.iter_mut() {
-                            bucket.update(selectivity);
-                        }
-                    }
+                if selectivity < 0.8 {
```

Comment:
`if selectivity < 0.2` ?

Reply:
If there is a little data filtered, assume the histogram is still valid.

Reply:
Why not update the histogram and keep it.

Reply:
The ndv will be approximated again and the histogram will be calculated based on the approximated ndv, the error in the histogram will increase, leading to an increased risk of -- -> +

Reply:
We can avoid this problem by pruning the buckets in the histogram according to the filter predicate, which means we need to implement a histogram whose number of buckets can be less than DEFAULT_HISTOGRAM_BUCKETS.

Reply:
Pruning and fewer buckets are not the key issue. The problem is that the min, max, ndv used to generate the histogram again are further approximated, and the histogram generation is also approximated, which leads to further errors in the lower bound, upper bound, and ndv of the new histogram's bucket being amplified. I don't see the need to regenerate the histogram at the moment, but if there is an actual case or job benchmark that has problems, I'll consider if I can address it further.

Reply:
For example, if there are 100 buckets and the predicate a < 50 can prune 50 buckets, we can continue to use the remaining 50 buckets instead of discarding them.

Reply:
Oh, I got you. It's a good way to maintain the **current column**'s part histogram buckets, no regeneration, directly to prune and replace. Let's do it later.